src/test/java/de/bsi/secvisogram/csaf_cms_backend/couchdb/CouchDbServiceTest.java

```diff
@@ -130,10 +133,66 @@ public void deleteCsafDocumentFromDb_invalidUuid() throws IOException {
         final UUID uuid = UUID.randomUUID();
         final String revision = insertTestDocument(uuid);
 
-        Assertions.assertThrows(DatabaseException.class,
+        assertThrows(DatabaseException.class,
                 () -> this.couchDbService.deleteCsafDocument("invalid user id", revision));
     }
 
+    @Test
+    @SuppressFBWarnings(value = "PRMC_POSSIBLY_REDUNDANT_METHOD_CALLS", justification = "document should not change")
+    public void bulkDeleteDocuments() throws IOException, DatabaseException {
+
+        long countBefore = this.couchDbService.getDocumentCount();
+
+        final UUID uuid1 = UUID.randomUUID();
+        String revision1 = insertTestDocument(uuid1);
+        final UUID uuid2 = UUID.randomUUID();
+        String revision2 = insertTestDocument(uuid2);
+
+        Assertions.assertEquals(countBefore + 2, this.couchDbService.getDocumentCount());
+        this.couchDbService.bulkDeleteDocuments(Arrays.asList(new IdAndRevision(uuid1.toString(), revision1),
+                new IdAndRevision(uuid2.toString(), revision2)));
+
+        Assertions.assertEquals(countBefore, this.couchDbService.getDocumentCount());
+    }
+
+    @Test
+    @SuppressFBWarnings(value = "PRMC_POSSIBLY_REDUNDANT_METHOD_CALLS", justification = "document should not change")
+    public void bulkDeleteDocuments_invalidUuid() throws IOException, DatabaseException {
+
+        long countBefore = this.couchDbService.getDocumentCount();
+
+        final UUID uuid1 = UUID.randomUUID();
+        String revision1 = insertTestDocument(uuid1);
+        final UUID uuid2 = UUID.randomUUID();
+        String revision2 = insertTestDocument(uuid2);
+
+        Assertions.assertEquals(countBefore + 2, this.couchDbService.getDocumentCount());
+        this.couchDbService.bulkDeleteDocuments(Arrays.asList(new IdAndRevision("Invalid uuid", revision1),
+                new IdAndRevision(uuid2.toString(), revision2)));
+
+        Assertions.assertEquals(countBefore + 1, this.couchDbService.getDocumentCount());
+    }
+
+    @Test
+    @SuppressFBWarnings(value = "PRMC_POSSIBLY_REDUNDANT_METHOD_CALLS", justification = "document should not change")
+    public void bulkDeleteDocuments_invalidRevision() throws IOException, DatabaseException {
+
+        long countBefore = this.couchDbService.getDocumentCount();
+
+        final UUID uuid1 = UUID.randomUUID();
+        String revision1 = insertTestDocument(uuid1);
+        final UUID uuid2 = UUID.randomUUID();
+        insertTestDocument(uuid2);
+
+        Assertions.assertEquals(countBefore + 2, this.couchDbService.getDocumentCount());
+        assertThrows(DatabaseException.class,
+                () -> this.couchDbService.bulkDeleteDocuments(Arrays.asList(
+                        new IdAndRevision(uuid1.toString(), revision1),
+                        new IdAndRevision(uuid2.toString(), "Invalid Revision"))));
+
+        Assertions.assertEquals(countBefore + 2, this.couchDbService.getDocumentCount());
```

Comment:
I'm wondering if it is expected behavior for bulk deletion to not delete any documents at all if one revision in the list is invalid. Comparing with `bulkDeleteDocuments_invalidUuid` above, it seems to delete other documents with valid id and revision still :thinking: